runtime/src/bank/tests.rs

```diff
@@ -12823,6 +12823,97 @@ fn test_rewards_computation() {
+        .map(|_| ValidatorVoteKeypairs::new_rand())
+        .collect::<Vec<_>>();
+
+    let GenesisConfigInfo { genesis_config, .. } = create_genesis_config_with_vote_accounts(
+        1_000_000_000,
+        &validator_keypairs,
+        vec![2_000_000_000; expected_num_delegations],
+    );
+
+    let bank0 = Bank::new_for_tests(&genesis_config);
+    let num_slots_in_epoch = bank0.get_slots_in_epoch(bank0.epoch());
+    assert_eq!(num_slots_in_epoch, 32);
+
+    // simulate block progress
+    let mut previous_bank = Arc::new(bank0);
+    for slot in 0..num_slots_in_epoch + 2 {
+        let pre_cap = previous_bank.capitalization();
+        let curr_slot = slot + 1;
+        let curr_bank = Bank::new_from_parent(&previous_bank, &Pubkey::default(), curr_slot);
+        let post_cap = curr_bank.capitalization();
+
+        // Fill banks with banks with votes landing in the next slot
+        // Create enough banks such that vote account will root
+        for validator_vote_keypairs in validator_keypairs.iter() {
+            let vote_id = validator_vote_keypairs.vote_keypair.pubkey();
+            let mut vote_account = curr_bank.get_account(&vote_id).unwrap();
+            // generate some rewards
+            let mut vote_state = Some(vote_state::from(&vote_account).unwrap());
+            for i in 0..MAX_LOCKOUT_HISTORY + 42 {
+                if let Some(v) = vote_state.as_mut() {
+                    vote_state::process_slot_vote_unchecked(v, i as u64)
+                }
+                let versioned = VoteStateVersions::Current(Box::new(vote_state.take().unwrap()));
+                vote_state::to(&versioned, &mut vote_account).unwrap();
+                match versioned {
+                    VoteStateVersions::Current(v) => {
+                        vote_state = Some(*v);
+                    }
+                    _ => panic!("Has to be of type Current"),
+                };
+            }
+            curr_bank.store_account_and_update_capitalization(&vote_id, &vote_account);
+        }
+
+        if curr_slot == num_slots_in_epoch {
+            // This is the first block of epoch 1. Reward computation should happen in this block.
+            // assert reward compute status activated at epoch boundary
+            assert!(matches!(
+                curr_bank.get_reward_interval(),
+                RewardInterval::InsideInterval
+            ));
+
+            // cap should increase because of new epoch rewards
+            assert!(post_cap > pre_cap);
+        } else if curr_slot == num_slots_in_epoch + 1 {
+            // This is the 2nd block of epoch 1. Reward distribution should happen in this block.
+            // assert stake rewards are paid at the first block after epoch boundary and the reward_status
+            // transitioned to inactive.
+            assert!(matches!(
+                curr_bank.get_reward_interval(),
+                RewardInterval::OutsideInterval
```

Comment:
yes. because there are no more pending rewards, the reward active status is cleared.

Reply:
@brooksprumo rewards are distributed early on in the block. Once we are in the last block to distribute rewards, the interval becomes OUTSIDE because we don't need to be inside anymore. So, even though the bank could be processing txs, there are no more rewards to distribute, so for the purposes of distributing rewards, we are now outside.